src/replica_ctrl.py

```diff
@@ -207,70 +222,77 @@ def update_config(primary_ip, current_ips, new_ips, replicaset_name, mongo_port)
 
```

Comment:
Got it. It seemed easier to apply your changes in "improvement" in this branch than the other way around.